src/lib/merkle_ledger/ledger.ml

```diff
@@ -404,5 +404,8 @@ end = struct
     let first_index = Addr.to_int a lsl height in
     let count = min (1 lsl height) (num_accounts t - first_index) in
     let subarr = Dyn_array.sub t.accounts first_index count in
-    Dyn_array.to_list subarr
+    List.zip_exn
```

Comment:
Can you do this in a safer way. Can you use `Addr.Range.fold` to iterate through all the addresses and then get the addresses? Like in `Database.ml`.